src/gbxml/ForwardTranslator.cpp

```diff
@@ -1350,6 +1327,40 @@ namespace gbxml {
     return result;
   }
 
+  boost::optional<pugi::xml_node> ForwardTranslator::translateId(const openstudio::model::ModelObject& modelObject, pugi::xml_node& parentElement) {
+    boost::optional<pugi::xml_node> result;
+
+    std::string id;
+    id = modelObject.name().get();
+    if (modelObject.hasAdditionalProperties()) {
+      model::AdditionalProperties additionalProperties = modelObject.additionalProperties();
+      if (additionalProperties.hasFeature("gbXMLId")) {
+        id = additionalProperties.getFeatureAsString("gbXMLId").get();
+      }
+    }
```

Comment:
You added a helper  `boost::optional<std::string> ModelObject_Impl::gbXMLId() const`, so use that instead.